homeassistant/components/switch/rest.py

```diff
@@ -36,26 +37,38 @@ def setup_platform(hass, config, add_devices_callback, discovery_info=None):
         _LOGGER.error("No route to resource/endpoint: %s", resource)
         return False
 
+    cfg_headers = config.get('headers')
+
+    if cfg_headers is None:
+        headers = DEFAULT_HEADERS
+    else:
+        headers = {}
+        for header in cfg_headers:
```

Comment:
The only thing I can think of that may cause an issue is when using hyphens in the header name when defining them:

```yaml
headers:
  Content-Type: text/plain
```

Is this valid YAML syntax?